Automattic/newspack-plugin
# Donation form processing

### All Submissions:






### Changes proposed in this Pull Request:



Blocked by #199. The relevant commit for this PR is https://github.com/Automattic/newspack-plugin/commit/f1087e0dafb3350fddd9bb2a6bba4a9160564806

This PR adds processing for the Donations block (or other forms that use compatible fields). When the form is submitted, the correct product gets added to the cart at the appropriate price and the user is redirected to the Checkout page.

### How to test the changes in this Pull Request:

1. Check out https://github.com/Automattic/newspack-blocks/pull/64 if it hasn't been merged yet. Set up your donation settings and insert a Donation block somewhere.
2. On the frontend, pick how you'd like to donate, and click the Donate Now button in the block. You should be dropped off on the Checkout page. Scroll down to order details and verify the order details match what you submitted. 

Checkout page styling will be in a future PR; the order details along with a link to change your selected donation will be at the top in the future. In the meantime, if you need to change your donation, go to `{siteURL}/cart` and remove it there.

<img width="640" alt="Screen Shot 2019-07-31 at 11 43 34 AM" src="https://user-images.githubusercontent.com/7317227/62238853-a7951400-b388-11e9-9189-ff447485769c.png">

### Other information:
